### ISSUE TITLE
Certificates errors and clusters can't be deployed in 2.4 HA setup

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#25284]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**

Single node 2.4 works fine. 2.3.5 HA also works as expected.

1. Deploy Rancher 2.4 HA using.  I've used commit id: 636311443
```
helm_v3 install rancher rancher-alpha/rancher --version 2.4.0-alpha1 --namespace cattle-system --set hostname=izaac-ha5.qa.rancher.space --set rancherImageTag=master-head
```

2. Login and try to deploy a one node cluster with all roles (or multi-node).

**Result:**
There will be certificate errors in the browser and the rancher agents.

And while provisioning a cluster the logs will show up:
```
[ERROR] cluster [c-sdf9s] provisioning: Failed to set up SSH tunneling for host [x.x.x.x]: Can't establish dialer connection: can not build dialer to [c-sdf9s:m-jtrhh]

Cluster must have at least one etcd plane host: failed to connect to the following etcd host(s) [x.x.x.x]
```

**Kubernetes Ingress Controller Fake Certificate** This certificate gets issued instead of the Rancher generated certificate.

**Other details that may be helpful:**

Chart version 2.4.0-alpha1

**Environment information**
- Rancher version : `2.4 master-head`
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Infrastructure Provider
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud

**Comment by izaac** [comment]
I've tested a new 2.4 HA install from today commit id: f01cca783

I was able to login without any issues to the UI, the self-signed certificate it's now named `cert-manager` as expected. 
And now I'm able to provision a cluster without the previously reported error.

The only difference I've noticed now compared with 2.3.5 is the certificate Issuer Name is `dynamiclistener-ca` where in 2.3.5 is `cattle-ca` 

Other than that doing a high level exploration didn't notice any other issues.
@ibrokethecloud
